src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -1224,14 +1240,34 @@ internal ReadOnlySpan<byte> ValueSpan
             }
         }
 
-        // TODO make public https://github.com/dotnet/runtime/issues/33388
-        internal static bool DeepEquals(JsonElement left, JsonElement right)
+        /// <summary>
+        /// Compares the values of two <see cref="JsonElement"/> values for equality, including the values of all descendant elements.
+        /// </summary>
+        /// <param name="element1">The <see cref="JsonElement"/> to compare.</param>
+        /// <param name="element2">The <see cref="JsonElement"/> to compare.</param>
+        /// <returns><see langword="true"/> if the two values are equal; otherwise <see langword="false"/>.</returns>
+        /// <remarks>
+        /// Deep equality of two JSON values is defined as follows:
+        /// <list type="bullet">
+        /// <item>JSON values of different kind are not equal.</item>
+        /// <item>JSON constants <see langword="null"/>, <see langword="false"/> and <see langword="true"/> only equal themselves.</item>
+        /// <item>JSON numbers are equal if and only if they have they have equivalent decimal representations, with no rounding being used.</item>
+        /// <item>JSON strings are equal if and only if they are equal using ordinal string comparison.</item>
```

Comment:
If the escaping has to be the same, then that probably should be called out.

This sentence makes it clear that "e" and "E" are different, but if the JSON value is `"e"` or `"\u0065"` those would both manifest as "e" when turned into a System.String, but I believe you said will count as not equals.

Reply:
It uses ordinal equality up to unescaping, effectively the same as the `ValueEquals` method.